### DISCUSSION TITLE
Setting color for custom brush pattern

### DISCUSSION THREAD

**Discussion by Waterloo** [fabricjs/fabric.js#7489] [Q&A]
Hi There,

I am trying to create a custom brush pattern using patternBrush, but when I try setting color for the brush the change is not reflected.

here is the Codepen example: https://codepen.io/jithin-airtory/pen/MWvBBPE

is there something I am doing wrong for setting the color of the brush ?

**Comment by ShaMan123** [comment]
There's a straight forward demo http://fabricjs.com/freedrawing
The is wrong:
```
texturePatternBrush.source = img;
```

**Reply by Waterloo** [reply]
I made the codepen based on the free drawing demo, can you tell me what is wrong with 

```
texturePatternBrush.source = img;
```
Because the exact same code is displayed on the free drawing demo

```
    var img = new Image();
    img.src = '../assets/honey_im_subtle.png';

    var texturePatternBrush = new fabric.PatternBrush(canvas);
    texturePatternBrush.source = img;
```

thank you

**Reply by ShaMan123** [reply]
You are loading an image.
You need to wait for it to finish before executing stuff with it
https://codepen.io/shaman123/pen/eYExXjw

You have other problems with Image or filters.
I don't know their origin.
Good luck
